tests/pyanaconda_tests/kickstart_dispatcher.py

```diff
@@ -0,0 +1,486 @@
+        # processing some more elements - firewall command
+        firewall_elements = elements.get_elements(commands=["firewall"], process=True)
+        self.assertEqual(set(elements.unprocessed_elements), set.difference(set(unprocessed_elements), set(firewall_elements)))
+        self.assertEqual(elements.unprocessed_elements, [element6, element7])
+        # now process everthing
+        elements.get_elements(commands=[], sections=[], addons=[], process=True)
+        self.assertEqual(elements.unprocessed_elements, [])
+
+        ### Dumping kickstart
+        dumped_ks = elements.get_kickstart_from_elements(elements.all_elements)
+        self.assertEqual(dumped_ks, expected_ks_content)
+
+class SplitKickstartParserTest(unittest.TestCase):
+
+    def setUp(self):
+        self._kickstart_flat = [kickstart_flat, None]
+        self._kickstart_include = [kickstart_include, kickstart_include_output]
+        self._kickstart_samples = [
+            self._kickstart_flat,
+            self._kickstart_include,
+        ]
+        self._flat_kickstarts_raising = [
+            kickstart_section_without_end,
+        ]
+
+    def simple_split_kickstart_parser_test(self):
+        """This test should demonstrate usage and output of the parser."""
+        ks_content = """
+%pre
+echo PRE
+%end
+network --device=ens3 --activate
+network --device=ens4 --activate
+%addon pony --fly=True
+%end
+firewall --enabled
+%addon scorched --planet=Earth
+nuke
+%end
+%post --nochroot --interpreter /usr/bin/bash
+echo POST1
+%end
+""".lstrip()
+
+        def get_expected_result(filename):
+            element1 = KickstartElement(("%pre",), "%pre\necho PRE\n%end\n", 1, filename)
+            element2 = KickstartElement(("network",), "network --device=ens3 --activate\n", 4, filename)
+            element3 = KickstartElement(("network",), "network --device=ens4 --activate\n", 5, filename)
+            element4 = KickstartElement(("%addon", "pony"), "%addon pony --fly=True\n%end\n", 6, filename)
+            element5 = KickstartElement(("firewall",), "firewall --enabled\n", 8, filename)
+            element6 = KickstartElement(("%addon", "scorched"), "%addon scorched --planet=Earth\nnuke\n%end\n", 9, filename)
+            element7 = KickstartElement(("%post",), "%post --nochroot --interpreter /usr/bin/bash\necho POST1\n%end\n", 12, filename)
+            return [element1, element2, element3, element4, element5, element6, element7]
+
+        filename = "ks.test.simple.cfg"
+        expected_result = get_expected_result(filename)
+
+        valid_sections = ["%pre", "%pre-install", "%post", "%onerror", "%traceback", "%packages", "%addon", "%anaconda"]
+
+        handler = returnClassForVersion()
+        ksparser = SplitKickstartParser(handler, valid_sections)
+
+        ### Reading kickstart from file
+
+        with open(filename, "w") as f:
+            f.write(ks_content)
+        result = ksparser.split(filename)
+        #os.remove(filename)
```

Comment:
Remove this completely I don't think there is a point on having code commented out.

Reply:
Actually I should uncomment it :)

Reply:
That is the solution too :)